packages/core/src/shared/fs/node/fs.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import fs, { FileSystem } from '../fs'
+import vscode from 'vscode'
+import _path from 'path'
+import _fsNode from 'fs/promises'
+import lockFile, { LockOptions } from 'proper-lockfile'
+import { ToolkitError } from '../../errors'
+
+/**
+ * FileSystem related methods that only work in Node.js
+ */
+export class NodeFileSystem {
+    private fsCommon: FileSystem = fs
+
+    protected constructor() {}
+    static #instance: NodeFileSystem
+    static get instance(): NodeFileSystem {
+        return (this.#instance ??= new NodeFileSystem())
+    }
+
+    /**
+     * Acquires a lock on the given file, then runs the given callback, releasing the lock when done.
+     *
+     * The reason this method is node specific is that we use the `mkdir()` method for the lock file,
```

Comment:
is this saying that lockFile.lock() is using mkdir internally? mkdir is not used in this file directly

Reply:
Will update, by "we" I meant the `proper-lockfile` module uses it internally. Does this work?